### PR TITLE
ui: `onclick` propagation event trigger condition updated and search fixes

### LINKED ISSUE #671: ui: unique value filter per column: drop-down contains empty field, and also null
I think the following screenshots show unintended behavior in various ways. Just leaving this here to hopefully save some time.

Screenshot from 2023-02-04 14-24-38
Screenshot from 2023-02-04 14-27-28

Found in current staging environment, `runs/96d6f941f20e456a9d60b68699121084/`

### LINKED ISSUE #708: ui: Drop-down filter columns still change the sort when used
This is a follow on to #613 when I click on a drop down to filter, the ordering of the column with the dropdown is changed. I know we fixed this for text-fields, but I'm seeing it for dropdowns too. 

For example, the reason column on the main page: https://conbench.ursa.dev/ (the screen shot doesn't show much other than the changed state, but in case it's helpful!)

image

We should fix this up along with any of the other column types. If #661 is going to come soon, I'm fine just doing that which should _also_ solve this problem. But if we're not going to work on that soon let's fix this bug soon.

### LINKED ISSUE #734: Search: free text search not enabled for `commit message col` staging env
Happen to notice search wasn't enabled due to a missed condition

### PR DESCRIPTION
**Pull request by sayantikabanik**
closes #708 
  - Event trigger handler added for `click` action 
  
closes #734 
  - condition updated (>=10)

closes #671 
  - error col text added -> adding it drop-down
  - minor UI fix : setting static width
  - search logic updated for dropdown

~I have to test the condition after the web-app execution comes though.
Related issue #709~

### CODE REVIEW

**Comment by sayantikabanik** [comment]
testing a few edge-cases

More on various event propagation handlers -https://www.carlrippon.com/stoppropagation-v-stopimmediatepropagation/

**Comment by sayantikabanik** [comment]
I tried testing most combinations , but definitely  not all of them. 
The `null` selection still feels buggy, apart from this rest should work as expected. 

UPDATE:

The bug for null value selection is hopefully now resolved.

File: conbench/templates/app.html [hunk 1]
```diff
@@ -157,26 +157,46 @@ <h4 class="modal-title"><span class="glyphicon glyphicon-trash submenu"></span>&
...
                 });
             mySelectList.append(
-                $(`<option value="${reset}">${reset}</option>`)
+                $(`<option value="All">${reset}</option>`)
             );
             this.cache("search").unique().sort().each(function(param) {
-                // check for multiple whitespaces and empty string
-                if (/^\s*$/.test(param) == false) {
+                //trim the whitespaces (setting to empty string) before appending to drop-down list
+                param = param.trim()
```
Comment (jgehrcke) [inline conbench/templates/app.html:186]: this trims leading and trailing whitespace, yes?

**Review by jgehrcke — APPROVED** [review]
:rocket: thank you for the detailed PR description. I think we should land this and then iterate further as we see problems/challenges arise!

File: conbench/templates/app.html (hunk 1 above)
Comment (sayantikabanik) [reply conbench/templates/app.html:186]: Yup :)
